scripts/travis/before_build.sh

```diff
@@ -18,7 +18,9 @@ SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
 OS=$("${SCRIPTPATH}"/../ostype.sh)
 ARCH=$("${SCRIPTPATH}"/../archtype.sh)
 
-echo "Building libsodium-fork..."
-make crypto/libs/${OS}/${ARCH}/lib/libsodium.a
+if [ ! -f crypto/libs/${OS}/${ARCH}/lib/libsodium.a ]; then
+  echo "Building libsodium-fork..."
+  make crypto/libs/${OS}/${ARCH}/lib/libsodium.a
```

Comment:
How would we update these binaries when we change libsodium?

Reply:
Circleci is caching the libsodum builds based on the md5 sums of all files in libsodium fork, so if it ever changes a new build will happen and the cache will get updated

```
    - run:
          name: Get libsodium md5
          command: |
            mkdir -p tmp
            find crypto/libsodium-fork -type f -exec openssl md5 "{}" + > tmp/libsodium.md5
      - save_cache:
          key: 'libsodium-fork-{{ .Environment.CIRCLE_STAGE }}-{{ checksum "tmp/libsodium.md5" }}'
          paths:
            - crypto/libs
```